src/serialize/serializer.rs

```diff
@@ -790,6 +792,10 @@ impl Serialize for PyObject {
             Str::new(self.ptr).serialize(serializer)
         } else if py_is!(ob_type, BYTES_TYPE) {
             Bytes::new(self.ptr).serialize(serializer)
+        } else if py_is!(ob_type, BYTEARRAY_TYPE) {
+            ByteArray::new(self.ptr).serialize(serializer)
+        } else if py_is!(ob_type, MEMORYVIEW_TYPE) {
+            MemoryView::new(self.ptr).serialize(serializer)
```

Comment:
Actually, It would be better to handle `bytearray` and `memoryview` objects in `serialize_unlikely`, as they are not common. And they should also be handled for dict keys (`DictKey` type below `PyObject`).

Reply:
Updated. Moved `bytearray` and `memoryview` before opt-based stuff in `serialize_unlikely`, can move further down if needed.